noxfile.py

```diff
@@ -79,26 +211,63 @@ def tests(session):
       - https://github.com/theacodes/nox/issues/260
 
     """
-    # Determine whether pytest is installed as part of
-    # our conda requirements.
-    pytest = Path(session.bin) / "pytest"
-    if not pytest.is_file():
-        # Determine the conda requirements yaml file.
-        fname = f"requirements/py{PY_VER.replace('.', '')}.yml"
+    artifact = get_iris_github_artifact(session)
+    python_version = session.python.replace(".", "")
+    requirements = f"requirements/py{python_version}.yml"
+
+    if artifact:
+        tmp_dir = Path(session.create_tmp())
+        artifact_dir = tmp_dir / "iris"
+        cwd = Path.cwd()
+        if not artifact_dir.is_dir():
+            session.run("git", "clone", IRIS_GITHUB, str(artifact_dir), external=True)
+        session.cd(artifact_dir)
+        session.run("git", "fetch", "origin", external=True)
+        session.run("git", "checkout", artifact, external=True)
+        session.cd(cwd)
+        iris_requirements = f"{artifact_dir}/requirements/ci/py{python_version}.yml"
+        yaml_requirements = combine_requirements(requirements, iris_requirements)
+        requirements = tmp_dir / "requirements.yml"
```

Comment:
Reusing the "requirements" variable name adds a little to the variable name confusion. Perhaps a name like "temp_requirements_location" or "final_requirements_location" would be clearer. Alternatively, the initial "requirements" could be renamed something like "local_requirements_location".